src/argilla/server/models/database.py

```diff
@@ -207,6 +207,12 @@ def __repr__(self):
             f"inserted_at={str(self.inserted_at)!r}, updated_at={str(self.updated_at)!r})"
         )
 
+    def vector_by_vector_settings(self, vector_settings: "VectorSettings") -> Union[Vector, None]:
+        for vector in self.vectors:
+            if vector.vector_settings_id == vector_settings.id:
+                return vector.value
+        return None
```

Comment:
Maybe instead of this method we should create inside the context file a `get_record_vector_for_vector_settings(record_id, vector_settings_ids)` or similar (doing only a query instead of iterating through all vectors collection).

Apart from that if we don't want to do that previous change we can remove the explicit `return None` and change the name of the method because we are not returning a vector but it's value instead:

```suggestion
    def vector_value_by_vector_settings(self, vector_settings: "VectorSettings") -> Union[List[float], None]:
        for vector in self.vectors:
            if vector.vector_settings_id == vector_settings.id:
                return vector.value
```

Reply:
Renamed. About moving to a context function, my feelings are that the current code base is not yet prepared for that. We're working with ORM classes in other parts than DB repositories. We should use agnostic domain classes shared by all the layers, feeding them from the proper places. 

For example, this specific logic is needed in the search engine module, which should be totally agnostic about the ORM classes